orangecontrib/spectroscopy/widgets/owoverlay.py

```diff
@@ -246,9 +248,14 @@ def set_vimage(self, data):
                     image_bytes=img_bytes,
                 )
 
-                # Assign it to the datatable attributes
-                if self.maindata and vimage is not None:
-                    self.maindata.attributes["visible_images"] = [vimage]
+                # # Assign it to the datatable attributes
+                if newmaindata and vimage is not None:
+                    if "visible_images" in list(newmaindata.attributes):
```

Comment:
`.attributes` is a dict, so conversion to list is not needed (and it actually makes it, O(n), while dict lookup is O(1); though that is not important here). So: `if "visible_images" in newmaindata.attributes:`